nx/guides/modoc_agg_nx_01.livemd

```diff
@@ -0,0 +1,826 @@
+\\ 
+y=1
+\begin{bmatrix}
+  & 1, 2, 3 & \\
+  & 1, -2, 3 &
+\end{bmatrix}
+\\ 
+y=2
+\begin{bmatrix}
+  & 4, -3, 2 & \\
+  & -4, 3, 2 &
+\end{bmatrix}
+```
+
+\end{bmatrix}
+\end{bmatrix}
+
+\to
+  \begin{bmatrix} x=0
+
+```
+\begin{bmatrix}
+  & \boxed{\rm{agg}(1,1)}, \rm{agg}(2,-2), \rm{agg}(3,3)& \\
+  & \rm{agg}(4,-4), \rm{agg}(-3,3), \rm{agg}(2,2) & \\
+  & \rm{agg}(5,-5), \rm{agg}(-1,1), \rm{agg}(3,3)
+\end{bmatrix}
+\\ 
+x=1
+\begin{bmatrix} 
+  & \rm{agg}(4,4), \boxed{\rm{agg}(6,-6)}, \rm{agg}(1,1) & \\
+  & \rm{agg}(1,1), \rm{agg}(2,-2), \rm{agg}(3,3) & \\
+  & \rm{agg}(4,-4), \rm{agg}(-3,3), \rm{agg}(2,2) &
+\end{bmatrix}
+```
+
+\end{bmatrix}
+
+\xrightarrow[]{\rm{argmax}}
+
+\begin{bmatrix}
+
+```
+\begin{bmatrix}
+  & 0, 0, 0 & \\
+  & 0, 1, 0 & \\
+  & 0, 1, 0
+\end{bmatrix}
+\\ 
+\begin{bmatrix} 
+  & 0, 0, 0 & \\
+  & 0, 0, 0 & \\
+  & 0, 1, 0
+\end{bmatrix}
+```
+
+\end{bmatrix}
+$$
+
+```elixir
+Nx.argmax(t, axis: :z)
+```
+
+###### Aggregate along the last axis, `axis: :t` or (`axis: 3` or `axis: -1`).
+
+This will reshape the tensor to `[x: 2, y: 3, z: 2]`. Since this is the last index, having in mind the matrix picture above, you aggregate along each row:
+
+$$
+r[x][y][z] = \rm{agg}\big(t[x][y][z]0,  t[x][y][z]1, t[x][y][z]2\big)
+$$
+
+and repeat this for each $x$-slice, each sub $y$-slice, each sub-sub $z$-slice.
+
+Below is the result for the function `argmin`.
+
+$$
+\begin{bmatrix} x=0
+  \begin{bmatrix} y=0
+
+```
+\begin{bmatrix}
+  & \boxed{1, 2, 3}, & \\
+  & 1, -2, 3, &
+\end{bmatrix}
+\\ 
+y=1
+\begin{bmatrix} 
+  & 4, -3, 2 & \\
+  & -4, 3, 2 &
+\end{bmatrix}
+\\ 
+y=2
+\begin{bmatrix}
+  & 5, -1, 3 &\\
+  & -5, 1, 3 &
+\end{bmatrix}
+```
+
+\end{bmatrix}
+  \\
+  x=1
+  \begin{bmatrix} y=0
+
+```
+\begin{bmatrix}
+  & 4, 6, 1 & \\
+  & 4,-6, 1 &
+\end{bmatrix}
+\\ 
+y=1
+\begin{bmatrix}
+  & 1, 2, 3 & \\
+  & \boxed{1, -2, 3} &
+\end{bmatrix}
+\\ 
+y=2
+\begin{bmatrix}
+  & 4, -3, 2 & \\
+  & -4, 3, 2 &
+\end{bmatrix}
+```
+
+\end{bmatrix}
+\end{bmatrix}
+
+\xrightarrow[]{\rm{argmin}}
+
+\begin{bmatrix}
+
+```
+\begin{bmatrix}
+  & \boxed{0}, 1 & \\
+  & 1, 0 & \\
+  & 1, 0
+\end{bmatrix}
+\\ 
+\begin{bmatrix} 
+  & 2, 1 & \\
+  & 0, \boxed{1} & \\
+  & 1, 0
+\end{bmatrix}
+```
+
+\end{bmatrix}
+$$
+
+```elixir
+Nx.argmin(t, axis: 3)
+```
+
+#### Option `:tie_break`
+
+You have the `:tie_break` option to decide how to operate with you have several occurences of the result. It defaults to `tie_break: :low`.
+
+```elixir
+t4 = ~V[2 0 0 0 1]
+
+%{
+  argmin_with_default: Nx.argmin(t4) |> Nx.to_number(),
+  argmin_with_tie_break_high_option: Nx.argmin(t4, tie_break: :high) |> Nx.to_number()
+}
+```
+
+#### Option `:keep_axis`
+
+Its default value is `false` but when this option set to `keep_axis: true`, you reshape the resulting tensor with a new shape where the dimension of the working axis to $1$. For example, `Nx.argmax(t, axis: :y, keep_axix: true)` will have a shape a `[x: 2,y: 1,z: 2, t: 3]`. This mean that:
```

Comment:
```suggestion
Its default value is `false` but when this option set to `keep_axis: true`, instead of being totally collapsed, the aggregation axis will be reduced to size $1$. For example, `Nx.argmin(t, axis: :y, keep_axis: true)` will have a shape `{2, 1, 2, 3}`, as seen below:
```